hyperopt/tests/test_spark.py

```diff
@@ -0,0 +1,434 @@
+import contextlib
+import logging
+import unittest
+import tempfile
+import time
+import shutil
+
+from six import StringIO
+
+from pyspark.sql import SparkSession
```

Comment:
shouldn't we explicitly require `pyspark` as dependency in `setup.py`?

Reply:
Discussed with Xiangrui offline: How about we add it as `extras_require` to make it optional?